src/components/ReleaseMonitor/ReleasesInNamespace.tsx

```diff
@@ -1,41 +1,199 @@
 import * as React from 'react';
 import { useK8sAndKarchResources } from '~/hooks/useK8sAndKarchResources';
-import { ReleaseGroupVersionKind, ReleaseModel } from '~/models';
-import { MonitoredReleaseKind } from '~/types';
+import { useK8sWatchResource } from '~/k8s';
+import {
+  ReleaseGroupVersionKind,
+  ReleaseModel,
+  ReleasePlanAdmissionGroupVersionKind,
+  ReleasePlanAdmissionModel,
+  ReleasePlanGroupVersionKind,
+  ReleasePlanModel,
+} from '~/models';
+import {
+  MonitoredReleaseKind,
+  ReleaseKind,
+  ReleasePlanAdmissionKind,
+  ReleasePlanKind,
+} from '~/types';
+import { MultipleNamespaceAdmissionsWatcher } from './MultipleNamespaceAdmissionsWatcher';
 
 type ReleasesInNamespaceProps = {
   namespace: string;
   onReleasesLoaded: (releases: MonitoredReleaseKind[]) => void;
   onError: (error: unknown) => void;
 };
 
+// Type-safe enhancement function
+const enhanceReleaseToMonitored = (
+  release: ReleaseKind,
+  releasePlans: ReleasePlanKind[],
+  allReleasePlanAdmissions: ReleasePlanAdmissionKind[],
+): MonitoredReleaseKind => {
+  let product = '';
+  let productVersion = '';
+  let rpa = '';
+
+  const releasePlan = releasePlans.find((rp) => rp.metadata.name === release.spec.releasePlan);
+
+  if (releasePlan) {
+    const rpaName =
+      releasePlan.metadata?.labels?.['release.appstudio.openshift.io/releasePlanAdmission'];
+    rpa = rpaName || '';
+
+    if (rpaName) {
+      const rpaObj = allReleasePlanAdmissions.find((a) => a.metadata.name === rpaName);
+      if (rpaObj) {
+        product = rpaObj.spec.data?.releaseNotes?.product_name;
+        productVersion = rpaObj.spec.data?.releaseNotes?.product_version;
+      }
+    }
+  }
+
+  return {
+    ...release,
+    product: product || undefined,
+    productVersion: productVersion || undefined,
```

Comment:
hmm, I believe you could pass these variables directly, no need to fallback to `undefined` 🤔 

```tsx
  return {
    ...release,
    product,
    productVersion,
    rpa,
  };
```